dashboard/lib/services/lti.rb

```diff
@@ -93,13 +93,23 @@ def self.lti_user_type(current_user, lti_integration, nrps_section)
       return ::User::TYPE_STUDENT
     end
 
-    def self.initialize_lti_student_from_nrps(client_id:, issuer:, nrps_member:)
+    def self.initialize_lti_user_from_nrps(client_id:, issuer:, nrps_member:)
       nrps_member_message = Policies::Lti.issuer_accepts_resource_link?(issuer) ? nrps_member[:message].first : nrps_member
+      email_address = get_claim(nrps_member_message, :email)
+      account_type = Policies::Lti.get_account_type(nrps_member[:roles])
+
       user = ::User.new
       user.provider = ::User::PROVIDER_MIGRATED
-      user.user_type = ::User::TYPE_STUDENT
       user.name = get_claim_from_list(nrps_member_message, Policies::Lti::STUDENT_NAME_KEYS)
-      user.family_name = get_claim(nrps_member_message, :family_name)
+
+      if account_type == ::User::TYPE_TEACHER && email_address.present?
```

Comment:
Why do we check for an email address here? I know we get email addresses from Canvas, but it isn't technically required for platforms to provide it, so I just wanted to clarify the reasoning for requiring it here.